runtime/ms-rest/lib/serialization.js

```diff
@@ -554,10 +554,6 @@ function _getPolymorphicMapperObjectVersion(mapper, object, objectName, mode) {
         `"${mapper.type.polymorphicDiscriminator[polymorphicPropertyName]}" is the ` +
         `polmorphicDiscriminator and is a required property.`);
     }
-    if (object[mapper.type.polymorphicDiscriminator[polymorphicPropertyName]] === null ||
-      object[mapper.type.polymorphicDiscriminator[polymorphicPropertyName]] === undefined) {
-      throw new Error(`No discriminator field "${mapper.type.polymorphicDiscriminator[polymorphicPropertyName]}" was found in "${objectName}".`);
```

Comment:
I would recommend not removing this check. Rather do the following:
```javascript
if (object[mapper.type.polymorphicDiscriminator[polymorphicPropertyName]] === null ||
  object[mapper.type.polymorphicDiscriminator[polymorphicPropertyName]] === undefined) {
  return mapper;
}
```
The reason being few lines down on line 561 
```javascript
indexDiscriminator = mapper.type.uberParent + '.' + object[mapper.type.polymorphicDiscriminator[polymorphicPropertyName]];
```
`indexDiscriminator` will be `<the value of the parent model>.undefined` and that would be weird.

We would eventually return the same mapper. So why not save the execution time and return the mapper early.

Does that make sense?